nixos/modules/services/desktops/pipewire/pipewire.nix

```diff
@@ -72,15 +74,75 @@ in {
           https://github.com/PipeWire/pipewire/blob/master/NEWS
         '';
       };
+
+      config = {
+        defaultConfig = mkOption {
+          default = true;
+          type = types.bool;
+          description = lib.mdDoc ''
+            Enable default configs for PipeWire.
+          '';
+        };
+        pipewire = mkOption {
+          type = json.type;
+          default = lib.importJSON "${pipewire-json-conf}/pipewire.conf";
+          defaultText = "{}";
+          description = lib.mdDoc ''
+            Configuration for PipeWire. For details see
+            https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Config-PipeWire
+          '';
+        };
+        client = mkOption {
+          type = json.type;
+          default = lib.importJSON "${pipewire-json-conf}/client.conf";
+          defaultText = "{}";
+          description = lib.mdDoc ''
+            Configuration for PipeWire Client. For details see
+            https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Config-client
+          '';
+        };
+        client-rt = mkOption {
+          type = json.type;
+          default = lib.importJSON "${pipewire-json-conf}/client-rt.conf";
+          defaultText = "{}";
+          description = lib.mdDoc ''
+            Configuration for PipeWire Client-rt. For details see
+            https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Config-client and
+            https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Config-ALSA
+          '';
+        };
+        jack = mkOption {
+          type = json.type;
+          default = lib.importJSON "${pipewire-json-conf}/jack.conf";
+          defaultText = "{}";
+          description = lib.mdDoc ''
+            Configuration for PipeWire Jack. For details see
+            https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Config-JACK
+          '';
+        };
+        pulse = mkOption {
+          type = json.type;
+          default = lib.importJSON "${pipewire-json-conf}/pipewire-pulse.conf";
+          defaultText = "{}";
+          description = lib.mdDoc ''
+            Configuration for PipeWire Pulse. For details see
+            https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Config-PulseAudio
+          '';
+        };
+        filter-chain = mkOption {
+          type = json.type;
+          default = {};
+          defaultText = "{}";
+          description = lib.mdDoc ''
+            Configuration for the pipewire filter-chain. For details see
+            https://gitlab.freedesktop.org/pipewire/pipewire/-/wikis/Filter-Chain
+          '';
+        };
+      };
     };
   };
 
   imports = [
-    (lib.mkRemovedOptionModule ["services" "pipewire" "config"] ''
```

Comment:
Are we api-compatible with the old options? I guess we could do without as this won't go into effect before the next release :)

Reply:
yes, except that i renamed pipewire-pulse to just pulse